medcat/utils/checkpoint.py

```diff
@@ -132,12 +175,57 @@ def _get_steps_and_count(file_path) -> Tuple[int, int]:
         file_name_parts = os.path.basename(file_path).split('-')
         return int(file_name_parts[1]), int(file_name_parts[2])
 
-    @staticmethod
-    def _load_metadata(dir_path: str) -> Optional[Dict]:
-        ckpt_file_paths = [os.path.abspath(os.path.join(dir_path, f)) for f in os.listdir(dir_path)]
-        metadata_file_paths = [f for f in ckpt_file_paths if os.path.isfile(f) and f.endswith("checkpoint-meta.json")]
-        if metadata_file_paths:
-            with open(metadata_file_paths[0]) as f:
-                return jsonpickle.decode(f.read())
-        else:
-            return None
+
+@dataclass
+class CheckpointConfig(object):
+    output_dir: str = "checkpoints"
+    steps: int = Checkpoint.DEFAULT_STEP
+    max_to_keep: int = Checkpoint.DEFAULT_MAX_TO_KEEP
+
+
+class CheckpointManager(object):
```

Comment:
Is this designed to be used across unsupervised / supervised cat training as well as MetaCAT / RelCAT training? I don't this is necessary right now as checkpointing is only really useful for unsupervised training, i.e. restore_cdb is only called in train, not train_supevised (which is want we want right now). But fine to leave in for extending to supervised cat training.

Reply:
I thought the outcome of the discussion was it is fine as long as it does not bluntly remove the folder. Alright, will remove this eye-watering method.